lsp/nls/src/server.rs

```diff
@@ -50,7 +50,7 @@ impl Server {
             definition_provider: Some(OneOf::Left(true)),
             references_provider: Some(OneOf::Left(true)),
             completion_provider: Some(CompletionOptions {
-                trigger_characters: Some(vec![]),
+                trigger_characters: Some(vec![".".to_owned()]),
```

Comment:
This `"."` refers to exactly the same character as the one in `completion.rs` on lines 83 and 142. They refer to "Hey, the user just typed `.` and would like completion". This is not a coincidence.

Yet currently, it is not trivial for the reader to identify if they refer to the same thing, or where this `"."` comes from. They might as well have completely different meanings.

There is a **domain** coupling between:
* `server.rs` registers the trigger `.`
* `completion.rs` handles the trigger `.`

As a rule of thumb, it is healthy to have static (type checked) coupling between things that have a domain coupling (i.e., things that **should be coupled**). It adds semantic and clarity (and a teeny tiny bit of safety, but this is negligible here).

As such, I suggest extracting into a constant here (or in a better-suited place), and refer to that constant in `completion.rs`.